### PR TITLE
update choice states (monster, item, state, etc.)

### PR DESCRIPTION
**Pull request by JaskRendix**
PR updates choice states.

First, I've removed all the magic numbers from the code and replaced them with constants at the top of the page, which makes everything much cleaner and easier to understand. I've also extracted a new method to declutter the init code, making it more modular and manageable. 

Additionally, I've fixed a bug that was causing issues with the monster info and journal info. Specifically, if a monster had the same evolution twice (like when it can evolve using two different items), it would show up twice, but that's now been addressed. 

Lastly, I've made some improvements to the `translated_dialog_choice`. If there are more than 13 elements, a scroll bar will now appear, which prevents the screen from crashing. The scroll bar is also capped at 80% of the screen height, so everything should be much more stable.

### CODE REVIEW

File: tuxemon/states/choice/choice_item.py [hunk 1]
```diff
@@ -14,6 +14,14 @@
 from tuxemon.menu.theme import get_theme
 
 ChoiceMenuGameObj = Callable[[], None]
+LENGHT_NAME_ITEM = 10
```
Comment (kerizane) [inline tuxemon/states/choice/choice_item.py]: typo

**Review by kerizane — CHANGES_REQUESTED** [review]

File: tuxemon/states/choice/choice_item.py [hunk 2]
```diff
@@ -14,6 +14,14 @@
 from tuxemon.menu.theme import get_theme
 
 ChoiceMenuGameObj = Callable[[], None]
+LENGTH_NAME_ITEM = 10
```
Comment (kerizane) [inline tuxemon/states/choice/choice_item.py:17]: This const seems to be unused.

Comment (kerizane) [reply tuxemon/states/choice/choice_item.py:17]: Same for the other files.

Comment (kerizane) [reply tuxemon/states/choice/choice_item.py:17]: Haven't checked if the other consts are used.

**Review by kerizane — CHANGES_REQUESTED** [review]

File: tuxemon/states/choice/choice_item.py (hunk 2 above)
Comment (JaskRendix) [reply tuxemon/states/choice/choice_item.py:17]: it was the only one unused

**Review by kerizane — APPROVED** [review]

**Pull request merged**